modules/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3ClientSettingsTests.java

```diff
@@ -207,12 +207,18 @@ public void testMaxConnectionsCanBeSet() {
         );
         assertThat(settings.get("default").maxConnections, is(ClientConfiguration.DEFAULT_MAX_CONNECTIONS));
         assertThat(settings.get("other").maxConnections, is(maxConnections));
-        ClientConfiguration defaultConfiguration = S3Service.buildConfiguration(settings.get("default"));
+        ClientConfiguration defaultConfiguration = S3Service.buildConfiguration(settings.get("default"), false);
         assertThat(defaultConfiguration.getMaxConnections(), is(ClientConfiguration.DEFAULT_MAX_CONNECTIONS));
-        ClientConfiguration configuration = S3Service.buildConfiguration(settings.get("other"));
+        ClientConfiguration configuration = S3Service.buildConfiguration(settings.get("other"), false);
         assertThat(configuration.getMaxConnections(), is(maxConnections));
 
         // the default appears in the docs so let's make sure it doesn't change:
         assertEquals(50, ClientConfiguration.DEFAULT_MAX_CONNECTIONS);
     }
+
+    public void testNonDefaultRetryPolicy() {
```

Comment:
Should we call thist `testStatelessDefaultRetryPolicy`?

Reply:
Sure pushed 5103ae4b655ed42dd69d895223053fb03efec6dc